btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -454,6 +461,65 @@ func (sew *StakingEventWatcher) watchForSpend(spendEvent *notifier.SpendEvent, t
 	)
 }
 
+func (sew *StakingEventWatcher) buildSpendingTxProof(spendingTx *wire.MsgTx) (*btcstakingtypes.InclusionProof, error) {
+	txHash := spendingTx.TxHash()
+	if len(spendingTx.TxOut) == 0 {
+		return nil, fmt.Errorf("stake spending tx has no outputs")
+	}
+	details, status, err := sew.btcClient.TxDetails(&txHash, spendingTx.TxOut[0].PkScript)
+	if err != nil {
+		return nil, err
+	}
+
+	if status != btcclient.TxInChain {
```

Comment:
I wonder whether we need this check i.e whether ` <-spendEvent.Spend:` return when tx is in the mempool or in chain. (eitherway better to have it just in case)